### DISCUSSION TITLE
Is there no sign up?

### DISCUSSION THREAD

**Discussion by zwl1619** [nextauthjs/next-auth#2413] [Help]
### Question 💬

I read the docs on [Client API](https://next-auth.js.org/getting-started/client), I didn't find the content about  "sign up".
Does it not support "sign up"?

### How to reproduce ☕️

null

### Contributing 🙌🏽

No, I am afraid I cannot help regarding this

**Comment by ToguyC** [comment]
Hi @zwl1619 !

Yes indeed there is no clear sign up option directly usable.
I think that is because next-auth use a lot of provider, and so cannot make all the different sign up options for all of these.

But now, you can add custom `authorizationParams` in the `Signin` function.
With this, you can specify customs params for each provider.
For example, I'm using Auth0 in a project, so for the signup I added `screen_hint: "signup"` in the `authorizationParams`.

I do not know for the other provider, but I think it is approximately the same.

I hope this will be useful !

**Reply by earthpfasoo** [reply]
can i get authorizationParams in signin callback?
